lightning/src/routing/router.rs

```diff
@@ -78,10 +83,80 @@ impl< G: Deref<Target = NetworkGraph<L>>, L: Deref, S: Deref, SP: Sized, Sc: Sco
 			&random_seed_bytes
 		)
 	}
+
+	fn create_blinded_payment_paths<
```

Comment:
Given we have the amount, we probably want to use something more like `lightning-invoice`'s `sort_and_filter_channels` for this method, at least eventually. Currently the sum of inbound capacity of the paths we're returning may not be sufficient, IIUC.

Reply:
> Given we have the amount, we probably want to use something more like `lightning-invoice`'s `sort_and_filter_channels` for this method, at least eventually.

Hmm... perhaps eventually. Unlike in `lightning-invoice`, these are meant to be used immediately, so the channels should be live at very least when going to pay. We may want to prefer channels based on their available liquidity, which would require a `collect` and `sort`. I was trying to avoid this given this action can be initiated by anyone sending an onion message.

> Currently the sum of inbound capacity of the paths we're returning may not be sufficient, IIUC.

Added the check against inbound capacity for now as that was indeed missing. Currently, requiring that every channel has enough. I suppose if this constraint cannot be met we could relax it to a fraction of the amount based on `MAX_PAYMENT_PATHS`?

Reply:
> I was trying to avoid this given this action can be initiated by anyone sending an onion message.

The action can be initiated by anyone only if this node has a public offer, is that right? 

Another thing to note is that if we include one of our channels that is public but doesn't have enough confirmations to be in the public network graph, we'll be revealing to our counterparty that we are the likely destination for the payment. This is something that `sort_and_filter_channels` takes care of, and based on the docs for the method it also takes other considerations into account that seem relevant here. This may not be a big concern for v1, though.

Reply:
> The action can be initiated by anyone only if this node has a public offer, is that right?

Yeah, true.

> 
> Another thing to note is that if we include one of our channels that is public but doesn't have enough confirmations to be in the public network graph, we'll be revealing to our counterparty that we are the likely destination for the payment.

Does `list_usable_channels` mostly avoid this? I believe `sort_and_filter_channels` doesn't assume all channels passed are usable. Wouldn't our counterparty know the payment is likely for us if the channel is unannounced as well?

> This is something that `sort_and_filter_channels` takes care of, and based on the docs for the method it also takes other considerations into account that seem relevant here. This may not be a big concern for v1, though.

Hmm... yeah there is some overlap there, but I think the scenarios are also different since the introduction nodes aren't our counterparties here whereas for `sort_and_filter_channels` the hints are. So the filtering and selection criteria aren't necessarily going to be the same as there.

Also, as mentioned elsewhere, we probably need to do some sorting after all so the payer knows which paths to prefer (e.g., ones where we have higher confidence that the hop from the introduction node has enough liquidity).

Reply:
> Does list_usable_channels mostly avoid this? I believe sort_and_filter_channels doesn't assume all channels passed are usable. 

I believe `list_usable_channels` just checks that the channel is funded, not how many confirmations it has. 

> Wouldn't our counterparty know the payment is likely for us if the channel is unannounced as well?

Yeah, so if we have public and private channels we should prefer the public ones in blinded payment paths we return, IIUC. 

> So the filtering and selection criteria aren't necessarily going to be the same as there.

With you there, just saying we'll probably want to see which of its checks are relevant for us when we add sorting.

Reply:
> I believe `list_usable_channels` just checks that the channel is funded, not how many confirmations it has.

AFAICT, it goes it is consider usable when both parties sent `channel_ready`, which happens once the funding transaction has enough confirmations.

> Yeah, so if we have public and private channels we should prefer the public ones in blinded payment paths we return, IIUC.

I'm not sure I understand why. They are blinded from the payer, so the concern we have with invoice hints doesn't apply. Maybe our counterparty learns something if we have both public and private channels with them and the payer uses MPP over both? Let me know if I'm missing something.

Reply:
> Maybe our counterparty learns something if we have both public and private channels with them and the payer uses MPP over both? Let me know if I'm missing something.

I was thinking that the counterparty can be fairly confident that we're the recipient if a private channel is used, but if public channels are used then we have more deniability. Privacy concerns aren't too high priority rn though, as discussed offline.